Fix NPE when calling ParsingException constructors with null reason

SirYwell opened the pull request:
<!--
Pull requests regarding major or minor updates need to target the `develop` branch.
Pull requests regarding patch updates need to target the `main` branch.
Please make sure to select the appropriate branch while creating the pull request.
For a reference on semantic versioning, see https://semver.org.
-->
A per the Javadocs, `null` values are permitted for the `reason`, however passing `null` or using a constructor that doesn't take a `String` causes a NullPointerException.

I addressed this issue and created a regression test for it.

Note: `String#isBlank` returns true if the string is empty, so this check wasn't needed.

tsaglam commented:
Your branch is based on main, but should be based on develop.

tsaglam reviewed (changes requested):
Your branch is based on main but should be based on develop.

SirYwell commented:
From my understanding of the comment in the pull request template https://github.com/jplag/JPlag/blob/main/.github/pull_request_template.md?plain=1#L3 together with semver (<https://semver.org/#summary>)
> PATCH version when you make backwards compatible bug fixes

I thought main would be the appropriate branch. I can change the branch, but I think the template is confusing then.

JanWittler commented on language-api/src/test/java/de/jplag/ParsingExceptionTest.java:
```diff
@@ -0,0 +1,23 @@
+public class ParsingExceptionTest {
+
+    @Test
+    void testParsingExceptionAcceptsNullReason() {
```
Just a small note but the test method does not match its body because you are testing far more than only a null reason.
`testParsingExceptionConstructor` maybe?

SirYwell replied:
I can probably find a better name, but I initially stumbled upon this because I called the `ParsingException(File, Throwable)` constructor. It delegates to the `ParsingException(File, String, Throwable)` constructor with `null` for the string, so the NPE was thrown even when not passing a reason at all. I decided to call all public constructors in the test case to cover internal delegations, but I could also only test the `ParsingException(File, String, Throwable)` constructor instead.

Nullability of the other arguments isn't tested here, so `testParsingExceptionConstructor` might be too generic. Maybe `testParsingExceptionAcceptsNoOrNullReason`?

JanWittler replied:
Ah thanks for the clarification. Actually, I would say in such a case it is sufficient to call the 3 args constructor if for the other constructors internal delegation is performed anyways. But it doesn't harm to have all 4 calls here neither, so then just leave it as is (and also the method name)

JanWittler reviewed (commented):
Just one smaller comment, besides that looks fine

SirYwell commented:
> I would still merge it into develop, as I am probably not releasing a patch version for this alone (I changed the target initially to develop).

I changed the target branch and rebased the commit onto develop.
